GEFS Staging in exglobal_stage_ic

AnilKumar-NOAA opened the pull request:
# Description
Adjusts source paths for forecast-only ICs in support of extension to ensembles. New paths omit the resolution (that is captured by the IC id name) and allow for a member level.

Directories with the reorganized ICs have been added to the three tier-1 machines under glopara space.

Building on this, also adds capability to stage GEFS ICs.

Resolves #911

# Type of change
- New feature

# Change characteristics
- Is this a breaking change (a change in existing functionality)? YES
- Does this change require a documentation update? NO

# How has this been tested?

# Checklist
- [ ] Any dependent changes have been merged and published
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] I have made corresponding changes to the documentation if necessary

AnilKumar-NOAA committed:
start stagging the code

AnilKumar-NOAA committed:
Add gefs selections and modified source and target in ATMOS INPUT

AnilKumar-NOAA committed:
remove comment line

AnilKumar-NOAA committed:
gefs sourcing and target updates

AnilKumar-NOAA committed:
updated code to read mem_dir in gefs

AnilKumar-NOAA committed:
update the gefs ensemble source dir

AnilKumar-NOAA committed:
updated source for the gfs_ctrl.nc

AnilKumar-NOAA committed:
Add Ensemble flag check

AnilKumar-NOAA committed:
Removed if block from gefs selection

AnilKumar-NOAA committed:
fixed loop in ATMOS sections

AnilKumar-NOAA committed:
assign value to ensemble members

AnilKumar-NOAA committed:
GEFS Stagging in ocean, ice, and wave

AnilKumar-NOAA committed:
GEFS Stagging with NSEM_ENS fixed

AnilKumar-NOAA committed:
Removing Shell checks

AnilKumar-NOAA committed:
Removing Shell checks

AnilKumar-NOAA committed:
Removing Shell checks

AnilKumar-NOAA committed:
Removing Shell checks

WalterKolczynski-NOAA commented:
Please fill in the PR template.

WalterKolczynski-NOAA commented on scripts/exglobal_stage_ic.sh:
```diff
@@ -8,6 +8,28 @@ GDATE=$(date -d "${PDY} ${cyc} - ${assim_freq} hours" +%Y%m%d%H)
 gPDY="${GDATE:0:8}"
 gcyc="${GDATE:8:2}"
 
+# Determine NMEM_ENS based on the configuration file (e.g., config.base)
+# Assuming NMEM_ENS is defined in config.base as an integer
+
+config_file="${EXPDIR}/config.base"
+
+if [[ -e "${config_file}" ]]; then
+    NMEM_ENS=$(grep "NMEM_ENS" "${config_file}" | awk -F= '{print $2}' || true)
+else
+    echo "Error: ${config_file} does not exist or is not accessible." >&2
+    exit 1
+fi
```
What is all this for? `config.base` is already sourced by every job, so NMEM_ENS will be set.

AnilKumar-NOAA replied:
ok. will remove it

WalterKolczynski-NOAA commented on scripts/exglobal_stage_ic.sh:
```diff
@@ -16,9 +38,21 @@ error_message(){
-# Start staging
+# Start staging gefs and gfs here
+
+# Check for command line argument to select gfs or gefs
+if [[ $# -ne 1 ]]; then
+  echo "Usage: ${0} [gfs | gefs]"
+  exit 1
+fi
+
 
+if [[ "${selection}" = "gfs" ]]; then
```
This should not be necessary, and is leading to a lot of duplication. Instead of these big if-blocks, there should be a small if-block based off of the `$RUN` that creates the list of `member_dirs` as you have above if it is `GEFS`, and otherwise makes it a one-element array with `""` (empty string) as the only entry. Once you've done this, you should simply loop over that for everything else and not need different blocks for GFS and GEFS.

AnilKumar-NOAA replied:
Is this way we have to OCN staging for gfs and gefs
if [[ "${DO_OCN:-}" = "YES" && "${RUN}" = "gfs" ]]; then

member_dir=""
NMEM_ENS=0
code here
elif [[ "${DO_OCN:-}" = "YES" && "${RUN}" = "gefs" ]]; then
code here (member_dir is already provided and NMEM_ENS as well so no need to define here?)

WalterKolczynski-NOAA replied:
No, you should only have one check of `$RUN` to make the memdir list. Everything after that is just looping through that list. You do not care whether it is GFS or GEFS once you have that list.

```
for MEMDIR in ${memdir[@]}; do
   # All the other code
done
```

WalterKolczynski-NOAA replied:
It's also important to use `MEMDIR`, as that is the variable that appears in the COM templates (see `config.com`). Otherwise the destinations won't be correct.

WalterKolczynski-NOAA commented on scripts/exglobal_stage_ic.sh:
```diff
@@ -8,6 +8,28 @@ GDATE=$(date -d "${PDY} ${cyc} - ${assim_freq} hours" +%Y%m%d%H)
+
+member_dirs=()
+
+# Populate the member_dirs array based on the value of NMEM_ENS
+for ((i = 0; i < "${NMEM_ENS}"; i++)); do
+    member_dirs+=("mem$(printf "%03d" "${i}")")
+done
```
See associated comment about not using big if-blocks or a command-line argument.
```suggestion
if [[ "${RUN}" == "gefs" ]]; then
  # Populate the member_dirs array based on the value of NMEM_ENS
  for ((i = 0; i < "${NMEM_ENS}"; i++)); do
      member_dirs+=("mem$(printf "%03d" "${i}")")
  done
else
  member_dirs+=("")
fi
```

AnilKumar-NOAA replied:
Updated

WalterKolczynski-NOAA commented on scripts/exglobal_stage_ic.sh:
```diff
@@ -16,9 +38,21 @@ error_message(){
+
+member_dir=""
+NMEM_ENS=0
+
```
```suggestion
```

WalterKolczynski-NOAA commented on scripts/exglobal_stage_ic.sh:
```diff
@@ -96,6 +192,21 @@ if [[ "${DO_WAVE:-}" = "YES" ]]; then
+  YMD=${gPDY} HH=${gcyc} generate_com -r COM_WAVE_RESTART
+  [[ ! -d "${COM_WAVE_RESTART}" ]] && mkdir -p "${COM_WAVE_RESTART}"
+    for member_dir in $(seq -w 0 $((NMEM_ENS - 1))); do
+    source="${BASE_CPLIC}/${CPL_ATMIC}/${YMD}${HH}/${member_dir}/ice/${PDY}.${cyc}0000.restart.gwes_30m"
```
gwes_30m is not a constant and depends on the wave grid being used.
```suggestion
    source="${BASE_CPLIC}/${CPL_ATMIC}/${YMD}${HH}/${member_dir}/ice/${PDY}.${cyc}0000.restart.${waveGRD}"
```

WalterKolczynski-NOAA reviewed (changes requested):
Didn't read a lot of it carefully because it is going to need a lot of rewriting.

`BASE_CPLIC` will also need to be updated on all the machines once I duplicate what I've done on Hera elsewhere.

AnilKumar-NOAA committed:
made modifications with =GFS

AnilKumar-NOAA committed:
Fixed

AnilKumar-NOAA committed:
fixed if loops for gfs and gefs

AnilKumar-NOAA committed:
looping over memdir for all staging components

AnilKumar-NOAA commented on scripts/exglobal_stage_ic.sh:
```diff
@@ -16,76 +28,75 @@
+# Start staging gefs and gfs here
 # Stage the FV3 initial conditions to ROTDIR (cold start)
+
+for MEMDIR in ${memdir[@]}; do
```
one this is not clear to me is that at gfs we have source target and gefs we have different path for source and target. here is the example and how we do this with one loop and we have member_dir loop for gefs under each compoenents to.
if [[ "${DO_WAVE:-}" = "YES" ]]; then
  member_dir=""
  NMEM_ENS=0
  YMD=${PDY} HH=${cyc} generate_com -r COM_WAVE_RESTART
  [[ ! -d "${COM_WAVE_RESTART}" ]] && mkdir -p "${COM_WAVE_RESTART}"
  for grdID in ${waveGRD}; do  # TODO: check if this is a bash array; if so adjust
    **source="${BASE_CPLIC}/${CPL_WAVIC}/${PDY}${cyc}/wav/${grdID}/${PDY}.${cyc}0000.restart.${grdID}"
    target="${COM_WAVE_RESTART}/${PDY}.${cyc}0000.restart.${grdID}"**
    ${NCP} "${source}" "${target}"
    rc=$?
    (( rc != 0 )) && error_message "${source}" "${target}" "${rc}"
    err=$((err + rc))
  done
elif [[ "${DO_WAVE:-}" = "YES" ]]; then
  YMD=${gPDY} HH=${gcyc} generate_com -r COM_WAVE_RESTART
  [[ ! -d "${COM_WAVE_RESTART}" ]] && mkdir -p "${COM_WAVE_RESTART}"
    **for member_dir in $(seq -w 0 $((NMEM_ENS - 1))); do
    source="${BASE_CPLIC}/${CPL_ATMIC}/${YMD}${HH}/${member_dir}/ice/${PDY}.${cyc}0000.restart.${waveGRD}"
    target="${COM_OCEAN_RESTART}/${PDY}.${cyc}0000.restart.${waveGRD}"
    ${NCP} "${source}" "${target}"**
    rc=$?
    (( rc != 0 )) && error_message "${source}" "${target}" "${rc}"
    err=$((err + rc))
    done
      echo "FATAL ERROR: Unsupported ocean resolution ${OCNRES}"
      rc=1
      err=$((err + rc))
fi

WalterKolczynski-NOAA replied:
For the target, it is baked into the COM variables through `MEMDIR`: https://github.com/NOAA-EMC/global-workflow/blob/e886e8aef633389d65e23cc42456827e1db2a771/parm/config/gfs/config.com#L47
https://github.com/NOAA-EMC/global-workflow/blob/e886e8aef633389d65e23cc42456827e1db2a771/parm/config/gfs/config.com#L52

For the source, you will need to add `MEMDIR` to the paths in the appropriate place. Since `MEMDIR` for the deterministic is just an empty string, it just becomes `//` in the path, which is treated the same as a single `/`.

AnilKumar-NOAA replied:
But we have to alteast use "for loop" on member_dir for all components or not in that case?

WalterKolczynski-NOAA replied:
All components should be within the for loop.

AnilKumar-NOAA replied:
I am talking about this section and member_dir loop here highlighted 
for MEMDIR in ${memdir[@]}; do

member_dir=""
NMEM_ENS=0
YMD=${PDY} HH=${cyc} generate_com -r COM_ATMOS_INPUT
[[ ! -d "${COM_ATMOS_INPUT}" ]] && mkdir -p "${COM_ATMOS_INPUT}"
  **for member_dir in $(seq -w 0 $((NMEM_ENS - 1))); do**
    source="${BASE_CPLIC}/${CPL_ATMIC}/${YMD}${HH}/${member_dir}/atmos/gfs_ctrl.nc"
    target="${COM_ATMOS_INPUT}/gfs_ctrl.nc"
    ${NCP} "${source}" "${target}"
    rc=$?
    (( rc != 0 )) && error_message "${source}" "${target}" "${rc}"
    err=$((err + rc))
    for ftype in gfs_data sfc_data; do
      for tt in $(seq 1 6); do
        source="${BASE_CPLIC}/${CPL_ATMIC}/${YMD}${HH}/${member_dir}/atmos/${ftype}.tile${tt}.nc"
        target="${COM_ATMOS_INPUT}/${ftype}.tile${tt}.nc"
        rc=$?
        (( rc != 0 )) && error_message "${source}" "${target}" "${rc}"
        err=$((err + rc))
      done
    done
  done

WalterKolczynski-NOAA replied:
That one is not correct. Use the one I gave you (and it needs to be before the `generate_com`).

AnilKumar-NOAA replied:
okay. Thanks Walter! I am learning a lot in shell scripting coding applications.

WalterKolczynski-NOAA replied:
I think it might be best if you started over from the original, then make just the following modifications
- Make the member_dirs list at the beginning as I recommended (like you now have it)
- Take the whole rest of the code and enclose it in the `for MEMDIR in ${member_dirs[@]}; do` for-loop
- Update the source paths to include `${MEMDIR}` where appropriate (destinations already have it in the COM templates)
- Update the file names to match what is in the alt_prototype directory where necessary (just the marine components, I think)

That should be all you have to do. Once you have your member_dirs list, you do not care whether it is GFS or GEFS, or how many members there are. All that information is encoded in the list.

AnilKumar-NOAA replied:
Please check the modifications. 
Let me fill up the template - any suggestions?

AnilKumar-NOAA committed:
memdir loop over all the components of exglobal_stage_ic

AnilKumar-NOAA committed:
fixed shellcheck warning for memdir loop

AnilKumar-NOAA committed:
Merge branch 'NOAA-EMC:develop' into upstream/develop

AnilKumar-NOAA committed:
completed patch for PR 1892

AnilKumar-NOAA committed:
Checked all indent issues

AnilKumar-NOAA committed:
Merge branch 'NOAA-EMC:develop' into upstream/develop

AnilKumar-NOAA committed:
Fixed indent issues for exglobal_stage_ic.sh

Fixed all indent issues. Please check

aerorahul commented on scripts/exglobal_stage_ic.sh:
```diff
@@ -8,99 +8,110 @@ GDATE=$(date --utc -d "${PDY} ${cyc} - ${assim_freq} hours" +%Y%m%d%H)
-      echo "FATAL ERROR: Unsupported ocean resolution ${OCNRES}"
-      rc=1
+  for ftype in gfs_data sfc_data; do
+    for tt in $(seq 1 6); do
```
```suggestion
    for ((tt = 1; tt <= 6; tt++)); do
```
To be consistent w/ the other loops in this script.

WalterKolczynski-NOAA commented on scripts/exglobal_stage_ic.sh:
```diff
@@ -8,99 +8,110 @@ GDATE=$(date --utc -d "${PDY} ${cyc} - ${assim_freq} hours" +%Y%m%d%H)
-  ${NCP} "${source}" "${target}"
+for MEMDIR in "${MEMDIR_ARRAY[@]}"; do
+  # Stage the FV3 initial conditions to ROTDIR (cold start)
+  YMD=${PDY} HH=${cyc} generate_com -r COM_ATMOS_INPUT
```
This will not work with the `-r` switch, which marks the variable as read-only. A second pass through the loop will fail. Same if true for all the `generate_com` within the loop.

AnilKumar-NOAA replied:
So in that case then removing -r is the option to go forward? 
We just implemented so I don't know whether this will work for both GFS and GEFS. Theoretically yes but need testing.
I see GEFS file conventional names differences with GFS input file. I'm working on GFS and GEFS out dir structure and files names and documenting.

WalterKolczynski-NOAA reviewed (changes requested):
Getting close, I think, but have you tested this for both GFS and GEFS?

AnilKumar-NOAA committed:
removed -r switch for all the generate_com

AnilKumar-NOAA commented:
> Getting close, I think, but have you tested this for both GFS and GEFS?

@WalterKolczynski-NOAA How to test on Hera?

WalterKolczynski-NOAA commented:
> > Getting close, I think, but have you tested this for both GFS and GEFS?
> 
> @WalterKolczynski-NOAA How to test on Hera?

To run GEFS, you're first going to need to take out the temporary hack Rahul made in [setup_expt.py](https://github.com/NOAA-EMC/global-workflow/blob/2d64b74a0b0407d01d6f657efa6af9637a131448/workflow/setup_expt.py#L237-L244) to stage GEFS for forecast only.

You also need to update the workflow/hosts file for the new `BASE_CPLIC` directory.

Then setup a forecast-only experiment for each.

AnilKumar-NOAA commented:
> > > Getting close, I think, but have you tested this for both GFS and GEFS?
> > 
> > 
> > @WalterKolczynski-NOAA How to test on Hera?
> 
> To run GEFS, you're first going to need to take out the temporary hack Rahul made in [setup_expt.py](https://github.com/NOAA-EMC/global-workflow/blob/2d64b74a0b0407d01d6f657efa6af9637a131448/workflow/setup_expt.py#L237-L244) to stage GEFS for forecast only.
> 
> You also need to update the workflow/hosts file for the new `BASE_CPLIC` directory.
> 
> Then setup a forecast-only experiment for each.

Shall i remove if loop or just section under gefs and keep it print statement?

 if inputs.system in ['gfs']:
        print('forecast-only mode treats ICs differently and cannot be staged here')
    elif inputs.system in ['gefs']:  # Temporarily copy ICs from icsdir into COM for      testing
        print('temporary hack to stage gefs ICs for testing')
      # comrot = os.path.join(inputs.comrot, inputs.pslot)
        #idatestr = datetime_to_YMDH(inputs.idate)
        #current_cycle_dir = f"gefs.{idatestr[:8]}"
       #cmd = f"cp -as {inputs.icsdir}/{current_cycle_dir} {comrot}/{current_cycle_dir}"
      #  os.system(cmd)
    return

WalterKolczynski-NOAA commented:
Remove the if-block, keeping just the first print message.

Will also need to add stage_ic as a job for GEFS. I will take care of that and make a PR to your branch.

AnilKumar-NOAA committed:
remove if condition and gefs manually staged condition in setup_expt.py

WalterKolczynski-NOAA committed:
Add stage_ic to GEFS job mesh and update dependencies

The structure of staged ICs have been updated to be more stream-
lined and also to allow ensemble members to support GEFS. The data
dependencies for the job are updated to use the new file structure.

The stage_ic job is also added to the GEFS job mesh to copy staged
initial conditions.

Refs #911

AnilKumar-NOAA commented:
You also need to update the workflow/hosts file for the new `BASE_CPLIC` directory.
in exglobal_script_ic
src="${BASE_CPLIC}/${CPL_ATMIC}/${YMD}${HH}/${MEMDIR}/atmos

We are setting hosts/hera.yaml 
BASE_CPLIC: '/scratch2/NCEPDEV/ensemble/noscrub/Walter.Kolczynski/global-workflow/alt_prototype_ICs/gefs_test'
So we need to put source 
src="${BASE_CPLIC}/${CPL_ATMIC}/${YMD}${HH}/${MEMDIR}/atmos/
And CPL_ATMIC is   parm/config/gfs/config.stage_ic we have labeled CPL_ATMIC - workflowtest
But this is under gfs folder. Do we have to change the folder to gefs? 

echo "BEGIN: config.stage_ic"
case "${CASE}" in
  "C48" | "C96")
    export CPL_ATMIC=workflowtest
    export CPL_ICEIC=workflowtest
    export CPL_OCNIC=workflowtest
    export CPL_WAVIC=workflowtest

WalterKolczynski-NOAA committed:
Add BASE_CPLIC to GEFS config.base

WalterKolczynski-NOAA committed:
Add stage_ic to GEFS config

WalterKolczynski-NOAA committed:
Remove stale coupled_ic config from GEFS

WalterKolczynski-NOAA committed:
Fix low-res staged IC directory names

WalterKolczynski-NOAA committed:
Add GEFS forecast dependency on stage_ic

WalterKolczynski-NOAA committed:
Fix indentation of print in setup_expt

WalterKolczynski-NOAA committed:
Load stage_ic config during exp setup for GEFS

WalterKolczynski-NOAA committed:
Change BASE_CPLIC location to temp new structure on Hera

The structure of staged ICs is being changed. For now, a temporary
directory with the new strucutre has been set up on Hera. This
changes `BASE_CPLIC` to that location for testing.

WalterKolczynski-NOAA commented:
Opened a PR into this branch: AnilKumar-NOAA/global-workflow#4

AnilKumar-NOAA committed:
Update rocoto generation for new stage_ic and add GEFS from WalterKolczynski-NOAA:feature/stage_gefs

AnilKumar-NOAA committed:
indent issue resolved

AnilKumar-NOAA committed:
indent issue resolved

AnilKumar-NOAA committed:
Merge pull request #4 from WalterKolczynski-NOAA/feature/stage_gefs

Update rocoto generation for new stage_ic and add GEFS

AnilKumar-NOAA committed:
Merge branch 'NOAA-EMC:develop' into upstream/develop

AnilKumar-NOAA committed:
gefs staging-exglobal_stage_ic.sh file is tested and code is running with no issues.

AnilKumar-NOAA committed:
Merge branch 'NOAA-EMC:develop' into upstream/develop

WalterKolczynski-NOAA committed:
Update prototype IC paths for refactored directories

AnilKumar-NOAA commented:
> Getting close, I think, but have you tested this for both GFS and GEFS?

Both GFS and GEFS settings the stage_ic running successfully on Hera.

AnilKumar-NOAA committed:
successful- tested coupled ic for gefs and gfs

AnilKumar-NOAA committed:
Merge branch 'upstream/develop' of https://github.com/AnilKumar-NOAA/global-workflow into upstream/develop

AnilKumar-NOAA committed:
Merge pull request #5 from WalterKolczynski-NOAA/feature/stage_gefs

Update prototype IC paths for refactored directories

AnilKumar-NOAA committed:
changes write permission to config.resource file

AnilKumar-NOAA committed:
config file permission to write for gfs

AnilKumar-NOAA committed:
undo permission for both gfs ad gefs

AnilKumar-NOAA committed:
undo permission for gfs/config.ocn

WalterKolczynski-NOAA committed:
Merge branch 'develop' into upstream/develop

WalterKolczynski-NOAA reviewed (approved):
Conditionally approved pending successful CI testing.

aerorahul commented on workflow/rocoto/gefs_tasks.py:
```diff
@@ -20,6 +75,8 @@ def fcst(self):
     def efcs(self):
         # TODO: Add real dependencies
         dependencies = []
+        dep_dict = {'type': 'task', 'name': f'{self.cdump}stage_ic'}
```
The `TODO` on L76 can go.  A dependency has been added.

aerorahul commented on scripts/exglobal_stage_ic.sh:
```diff
@@ -8,99 +8,89 @@ GDATE=$(date --utc -d "${PDY} ${cyc} - ${assim_freq} hours" +%Y%m%d%H)
+    ${NCP} "${src}" "${tgt}"
     rc=$?
-    (( rc != 0 )) && error_message "${source}" "${target}" "${rc}"
+    [[ ${rc} -ne 0 ]] && error_message "${src}" "${tgt}" "${rc}"
```
why `[[ ${rc} -ne 0 ]]` here? 
Elsewhere we have `(( rc != 0 ))`

aerorahul reviewed (commented):
some comments

aerorahul committed:
Update scripts/exglobal_stage_ic.sh

aerorahul commented on parm/config/gefs/config.waveinit:
```diff
@@ -0,0 +1,14 @@
+#! /usr/bin/env bash
```
Why is this being added?  Are we running with waves in GEFS at this point?

AnilKumar-NOAA replied:
we just copy that files from gfs when we were trying to run with wave.

aerorahul commented on scripts/exglobal_stage_ic.sh:
```diff
@@ -8,99 +8,89 @@ GDATE=$(date --utc -d "${PDY} ${cyc} - ${assim_freq} hours" +%Y%m%d%H)
+    ${NCP} "${src}" "${tgt}"
     rc=$?
-    (( rc != 0 )) && error_message "${source}" "${target}" "${rc}"
+    [[ ${rc} -ne 0 ]] && error_message "${src}" "${tgt}" "${rc}"
```
```suggestion
    ((rc != 0)) && error_message "${src}" "${tgt}" "${rc}"
```

aerorahul committed:
Update scripts/exglobal_stage_ic.sh

aerorahul committed:
Delete parm/config/gefs/config.waveinit

aerorahul commented on workflow/rocoto/gefs_tasks.py:
```diff
@@ -20,6 +75,8 @@ def fcst(self):
     def efcs(self):
         # TODO: Add real dependencies
```
```suggestion
```

aerorahul committed:
Update workflow/rocoto/gefs_tasks.py

TerrenceMcGuinness-NOAA commented:
```
(gfs_workflow) mterry (Orion-login-3) C48_ATM_a2716750 $ pwd
/work2/noaa/stmp/GFS_CI_ROOT/PR/1892/RUNTESTS/EXPDIR/C48_ATM_a2716750
```
```
(gfs_workflow) mterry (Orion-login-3) C48_ATM_a2716750 $ rocotocheck -d C48_ATM_a2716750.db -w C48_ATM_a2716750.xml -c 202103231200 -t gfsstage_ic

Task: gfsstage_ic
  account: nems
  command: /work2/noaa/stmp/GFS_CI_ROOT/PR/1892/global-workflow/jobs/rocoto/stage_ic.sh
  cores: 1
  cycledefs: gfs
  final: false
  jobname: C48_ATM_a2716750_gfsstage_ic_12
  join: /work2/noaa/stmp/GFS_CI_ROOT/PR/1892/RUNTESTS/COMROT/C48_ATM_a2716750/logs/2021032312/gfsstage_ic.log
  maxtries: 2
  name: gfsstage_ic
  nodes: 1:ppn=1:tpp=1
  partition: orion
  queue: batch
  throttle: 9999999
  walltime: 00:15:00
  environment
    CDATE ==> 2021032312
    CDUMP ==> gfs
    COMROOT ==> /work/noaa/global/glopara/com
    DATAROOT ==> /work/noaa/stmp/mterry/RUNDIRS/C48_ATM_a2716750
    EXPDIR ==> /work2/noaa/stmp/GFS_CI_ROOT/PR/1892/RUNTESTS/EXPDIR/C48_ATM_a2716750
    HOMEgfs ==> /work2/noaa/stmp/GFS_CI_ROOT/PR/1892/global-workflow
    NET ==> gfs
    PDY ==> 20210323
    RUN ==> gfs
    RUN_ENVIR ==> emc
    cyc ==> 12
  dependencies
    AND is not satisfied
      /work/noaa/global/glopara/data/ICSDIR/prototype_ICs/workflow_C48_refactored/2021032312/atmos/gfs_ctrl.nc does not exist

Cycle: 202103231200
  Valid for this task: YES
  State: active
  Activated: 2023-10-27 18:12:52 UTC
  Completed: -
  Expired: -

Job: This task has not been submitted for this cycle

Task can not be submitted because:
  Dependencies are not satisfied
```
```
(gfs_workflow) mterry (Orion-login-3) C48_ATM_a2716750 $ ls -l /work/noaa/global/glopara/data/ICSDIR/prototype_ICs/workflow_C48_refactored/2021032312/atmos/gfs_ctrl.nc
lrwxrwxrwx 1 wkolczyn global 49 Oct 19 14:33 /work/noaa/global/glopara/data/ICSDIR/prototype_ICs/workflow_C48_refactored/2021032312/atmos/gfs_ctrl.nc -> workflowtest/2021032312/gfs/C48/INPUT/gfs_ctrl.nc
(gfs_workflow) mterry (Orion-login-3) C48_ATM_a2716750 $ 
```
Not sure whey Rocoto is not finding this link to this `gfs_crl.nc` file.

JessicaMeixner-NOAA commented:
@aerorahul @WalterKolczynski-NOAA  - @jiandewang just pointed out that the IC folders have been moved from the ufscpara and his space on wcoss2.  We specifically moved it out of the glopara space on wcoss2 because of our data needs exceeding the space requirements on WCOSS2.   Is this intentional?  Did you just need us to refactor the space?  We'd be happy to work with you on this.  We will soon be adding ICs for HR3 and will need to know what the process for this is now.
